src/networking/VoiceUDPSocket.ts

```diff
@@ -74,16 +77,27 @@ export class VoiceUDPSocket extends EventEmitter {
 	}
 }
 
+const IPV4_REGEX = new RegExp(/^((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\.(?!$)|$)){4}$/);
+
 /**
  * Parses the response from Discord to aid with local IP discovery.
  *
  * @param message - The received message
  */
-function parseLocalPacket(message: Buffer): SocketConfig {
+export function parseLocalPacket(message: Buffer): SocketConfig {
 	// todo, this function can throw
 	const packet = Buffer.from(message);
 	let ip = '';
 	for (let i = 4; i < packet.indexOf(0, i); i++) ip += String.fromCharCode(packet[i]);
 	const port = parseInt(packet.readUIntLE(packet.length - 2, 2).toString(10), 10);
+
+	if (!IPV4_REGEX.exec(ip)) {
+		throw new Error('Malformed IP address');
+	}
```

Comment:
Also, this line could be moved above 92, so if the IP fails, it doesn't extract the port.
```suggestion
	if (!IPV4_REGEX.test(ip)) {
		throw new Error('Malformed IP address');
	}
```